### ISSUE TITLE
No longer resolves nested packages unless full path is provided

### ISSUE THREAD

**Issue by dtothefp** [martpie/next-transpile-modules#233]
- [x] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE
- [x] I WILL GIVE 10$ TO CHARITY IF MY SOLUTION WAS ACTUALLY IN THE README/FAQ

**Are you trying to transpile a local package or an npm package?**
NPM package. Multiple, for example `escape-string-regexp` which has non es5 compliant code

**Describe the bug**
Upgraded from 4.x to 8.x 
- used to just specify `@react-aria` which would compile all packages in the namespace. because `@react-aria` is just a namespace and not a package got a [module require error ](https://github.com/martpie/next-transpile-modules/blob/master/src/next-transpile-modules.js#L142). I'm assuming this is not just the functionality so not really a bug. I wrote a function to get the paths of all modules within `node_modules/@react-aria` and this was resolved
- secondly, this new resolution caused us a bug in IE because a package in our package.json required a different version of `escape-string-regexp` than was specified in the root package.json. Therefore, we had to supply all the potential paths to this package, ie
```js
const withTM = require('next-transpile-modules')([
  'escape-string-regexp',
  'mdast-util-find-and-replace/node_modules/escape-string-regexp'
]);

module.exports = withTM({
  webpack5: true,
});
```
This doesn't seem like a very sustainable config. Is there a reason that the plugin no longer compiles just based upon a match of the module name rather than the full path?

**To Reproduce**
Steps to reproduce the behavior (a reproduction is even better):

in package.json

```json
"dependencies": {
  "escape-string-regexp": "1.0.5",
  "mdast-util-find-and-replace": "1.1.1"
}
```

only use `escape-string-regexp` in your transpile modules config and check the output that only version one version of the lib is being compiled. the nested lib in `mdast` with version `^4.0.0` will remain uncompiled with es6 syntax

**Expected behavior**
All versions of `escape-string-regexp` in `node_modules` are compiled

**Setup**

- Next.js version: 11.1.0
- `next-transpile-modules` version:
- Node.js version: 
- `npm`/`yarn` version: yarn 1.22.10
- Operating System: OSX
- Webpack 4 or 5: 5

**Additional context**

**Comment by martpie** [comment]
So it's not really a bug and that's the intended behavior regarding having to specify the whole path. If your package manager nested a duplicate version of `escape-string-regexp` inside a dependency's `node_modules`, that probably means you have multiple versions of this library in your `node_modules`.

You may want to use (for example with Yarn) the `resolutions` field of your app's `package.json` (be careful to not use incompatible versions though), or solve the issue with `mdast-util-find-and-replace`'s maintainer.

https://classic.yarnpkg.com/en/docs/selective-version-resolutions/

Hope that helps!

**Comment by dtothefp** [comment]
@martpie I agree this is bad on our end to be including multiple versions of the same package but it's a big app. is there a reason for this change because it seemed like this "just worked' in previous versions

**Comment by martpie** [comment]
I'm pretty sure the problem does not come from ntm but from a dependencies update you may have done recently and that provocated this duplication of this package :) 

Did you happen to do that?

**Comment by Fi1osof** [comment]
It's work with next-transpile-modules@^4.0, but not next-transpile-modules@^9.0 :(((

**Comment by martpie** [comment]
Please read the changelogs

**Comment by Fi1osof** [comment]
> Please read the changelog :)

I tried, but find nothing helps me.

**Comment by martpie** [comment]
Please open a new issue 👍
